### PR TITLE
Crystal ball transfer functions

### PR DESCRIPTION
**Pull request by TomasDado**
This PR adds new parametrisation for TFs that uses crystal ball functions

## Description
Adding Crystal Ball function parametrisation for ljets

## How Has This Been Tested?
For now no tests have been done, hence the WIP status

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: include/KLFitter/ResCrystalBallBase.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+  /**
+    * A constructor.
+    * @param parameters The parameters of the parameterization.
+    */
+  explicit ResCrystalBallBase(std::vector<double> const& parameters);
+
+  /**
+    * The (defaulted) destructor.
+    */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResCrystalBallBase.h]: really? this is a problem?

File: include/KLFitter/ResSingleGaussMET.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  * \brief A class describing a resolution parameterized with a double Gaussian.
+  *
+  * This class offers a simple parameterization of a resolution. The
+  * parameterization is a double Gaussian with energy dependent
+  * parameters.
+  */
+class ResSingleGaussMET : public ResSingleGaussLinearBase {
+ public:
+  /** \name Constructors and destructors */
+  /* @{ */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResSingleGaussMET.h]: I am not sure what you mean

File: include/KLFitter/DetectorAtlas_CrystalBall.h [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+// ---------------------------------------------------------
+
+namespace KLFitter {
+class ResolutionBase;
+
+/**
+ * A class for describing of the ATLAS detector. This class
+ * holds the description of the ATLAS detector.
+ */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/DetectorAtlas_CrystalBall.h]: done

File: include/KLFitter/ResCrystalBallBase.h [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+#include <vector>
+
+#include "KLFitter/ResolutionBase.h"
+
+// ---------------------------------------------------------
+
+/**
+ * \namespace KLFitter
+ * \brief The KLFitter namespace
+ */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResCrystalBallBase.h]: done

File: include/KLFitter/ResCrystalBallBase.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResCrystalBallBase.h]: done

File: include/KLFitter/ResCrystalBallBase.h [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  /* @{ */
+
+  /**
+    * The default constructor.
+    */
+  explicit ResCrystalBallBase(const char * filename);
+
+  /**
+    * A constructor.
+    * @param parameters The parameters of the parameterization.
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResCrystalBallBase.h]: done

File: include/KLFitter/ResSingleGaussPt.h [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+#include <vector>
+
+#include "KLFitter/ResSingleGaussBase.h"
+
+// ---------------------------------------------------------
+
+/**
+ * \namespace KLFitter
+ * \brief The KLFitter namespace
+ */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResSingleGaussPt.h]: done

File: include/KLFitter/ResCrystalBallBase.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (knutzk) [reply include/KLFitter/ResCrystalBallBase.h]: No, it's not. But I started using this short-hand notation in other classes.

File: include/KLFitter/ResSingleGaussPt.h [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  * \brief A class describing a resolution parameterized with a double Gaussian.
+  *
+  * This class offers a simple parameterization of a resolution. The
+  * parameterization is a double Gaussian with energy dependent
+  * parameters.
+  */
+class ResSingleGaussPt : public ResSingleGaussBase {
+ public:
+  /** \name Constructors and destructors */
+  /* @{ */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResSingleGaussPt.h]: not sure what you mean

File: include/KLFitter/ResSingleGaussPt.h [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+ */
+namespace KLFitter {
+/**
+  * \class KLFitter::ResDoubleGaussE_1
+  * \brief A class describing a resolution parameterized with a double Gaussian.
+  *
+  * This class offers a simple parameterization of a resolution. The
+  * parameterization is a double Gaussian with energy dependent
+  * parameters.
+  */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResSingleGaussPt.h]: done

File: include/KLFitter/ResSingleGaussMET.h [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+ */
+namespace KLFitter {
+/**
+  * \class KLFitter::ResDoubleGaussMET
+  * \brief A class describing a resolution parameterized with a double Gaussian.
+  *
+  * This class offers a simple parameterization of a resolution. The
+  * parameterization is a double Gaussian with energy dependent
+  * parameters.
+  */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResSingleGaussMET.h]: done

File: include/KLFitter/ResSingleGaussMET.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (knutzk) [reply include/KLFitter/ResSingleGaussMET.h]: Remove these two lines, and the closing curly bracket a couple of lines below.

File: include/KLFitter/ResCrystalBallBase.h [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+ */
+namespace KLFitter {
+/**
+  * \class KLFitter::ResCrystalBallBase
+  * \brief A class describing a resolution parameterized with a double Gaussian.
+  *
+  * This class offers a simple parameterization of a resolution. The
+  * parameterization is a CrystalBall function with energy dependent
+  * parameters.
+  */
```
[the remark this answers is not in the document]
Comment (TomasDado) [reply include/KLFitter/ResCrystalBallBase.h]: I couldn't find Double gaussian mentioned in this header. although it was mentioned in other headers.

File: include/KLFitter/ResSingleGaussPt.h (hunk 7 above)
[the remark this answers is not in the document]
Comment (knutzk) [reply include/KLFitter/ResSingleGaussPt.h]: Remove these two lines, and the closing curly bracket a couple of lines below.

File: include/KLFitter/ResCrystalBallBase.h (hunk 10 above)
[the remark this answers is not in the document]
Comment (knutzk) [reply include/KLFitter/ResCrystalBallBase.h]: It is a couple of lines above in the `\name` and `\brief` tags I think.

**Comment by knutzk** [comment]
Thanks! Can you cherry-pick this PR into `dev-2.0`?

**Pull request merged**